ui/cypress/e2e/tokens.test.ts

```diff
@@ -0,0 +1,388 @@
+        {
+          id: id,
+          description: 'token test 02',
+          status: 'inactive',
+          permissions: [
+            {action: 'write', resource: {type: 'views'}},
+            {action: 'write', resource: {type: 'documents'}},
+            {action: 'write', resource: {type: 'dashboards'}},
+            {action: 'write', resource: {type: 'buckets'}},
+          ],
+        },
+        {
+          id: id,
+          description: 'token test 03',
+          status: 'inactive',
+          permissions: [
+            {action: 'read', resource: {type: 'views'}},
+            {action: 'read', resource: {type: 'documents'}},
+            {action: 'read', resource: {type: 'dashboards'}},
+            {action: 'read', resource: {type: 'buckets'}},
+          ],
+        },
+      ]
+
+      //check out array.reduce for the nested calls here
+      cy.request('api/v2/authorizations').then(resp => {
+        expect(resp.body).to.exist
+        authData.push({
+          description: resp.body.authorizations[0].description,
+          status: resp.body.authorizations[0].status === 'active',
+          id: resp.body.authorizations[0].id,
+        })
+
+        cy.createToken(
+          testTokens[0].id,
+          testTokens[0].description,
+          testTokens[0].status,
+          testTokens[0].permissions
+        )
+          .then(resp => {
+            expect(resp.body).to.exist
+            authData.push({
+              description: resp.body.description,
+              status: resp.body.status === 'active',
+              id: resp.body.id,
+            })
+          })
+          .then(() => {
+            cy.createToken(
+              testTokens[1].id,
+              testTokens[1].description,
+              testTokens[1].status,
+              testTokens[1].permissions
+            )
+              .then(resp => {
+                expect(resp.body).to.exist
+                authData.push({
+                  description: resp.body.description,
+                  status: resp.body.status === 'active',
+                  id: resp.body.id,
+                })
+              })
+              .then(() => {
+                cy.createToken(
+                  testTokens[2].id,
+                  testTokens[2].description,
+                  testTokens[2].status,
+                  testTokens[2].permissions
+                ).then(resp => {
+                  expect(resp.body).to.exist
+                  authData.push({
+                    description: resp.body.description,
+                    status: resp.body.status === 'active',
+                    id: resp.body.id,
+                  })
+                })
+              })
+              .then(() => {
+                //sanity check that data is seeded
+                cy.get<Organization>('@org').then(({id}) => {
+                  cy.request('api/v2/authorizations?orgID=' + id).then(resp => {
+                    expect(resp.body.authorizations).to.have.length(4)
```

Comment:
I don't think this sanity check is necessary.  This would also eliminate the need to chain all 3 of these create calls in nested `then` blocks.

Reply:
The check has been removed and the seeding block replaced with a loop 
```javascript
    testTokens.forEach(token => {
          cy.createToken(
            token.id,
            token.description,
            token.status,
            token.permissions
          ).then(resp => {
            expect(resp.body).to.exist
            authData.push({
              description: resp.body.description,
              status: resp.body.status === 'active',
              id: resp.body.id,
            })
          })
        })
```
Also the `cy.visit` call into the UI has been placed below this loop.  So it runs stably now headless and in Chrome.